### ISSUE TITLE
tailscaled tries to do UPnP/SSDP/hairpin check three times every second

### ISSUE THREAD

**Issue by hexchain** [tailscale/tailscale#4237]
### What is the issue?

My local network does not have UPnP or NAT-PMP. Here is a `tcpdump` on the machine with Tailscale running, it stops if I stop tailscaled:
```
% sudo tcpdump -ni eno1 'udp port (1900 or 12345 or 5351)'
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eno1, link-type EN10MB (Ethernet), snapshot length 262144 bytes
19:37:54.555638 IP 192.168.40.39.43824 > 203.0.113.1.12345: UDP, length 73
19:37:54.555729 IP 192.168.40.39.54925 > 192.168.40.1.5351: UDP, length 2
19:37:54.555764 IP 192.168.40.39.54925 > 192.168.40.1.5351: UDP, length 24
19:37:54.555786 IP 192.168.40.39.54925 > 192.168.40.1.1900: UDP, length 94
19:37:54.555808 IP 192.168.40.39.54925 > 239.255.255.250.1900: UDP, length 94
19:37:54.807934 IP 192.168.40.39.53716 > 203.0.113.1.12345: UDP, length 73
19:37:54.808012 IP 192.168.40.39.47555 > 192.168.40.1.5351: UDP, length 2
19:37:54.808035 IP 192.168.40.39.47555 > 192.168.40.1.5351: UDP, length 24
19:37:54.808051 IP 192.168.40.39.47555 > 192.168.40.1.1900: UDP, length 94
19:37:54.808067 IP 192.168.40.39.47555 > 239.255.255.250.1900: UDP, length 94
19:37:55.299887 IP 192.168.40.39.35289 > 203.0.113.1.12345: UDP, length 73
19:37:55.299972 IP 192.168.40.39.55225 > 192.168.40.1.5351: UDP, length 2
19:37:55.299997 IP 192.168.40.39.55225 > 192.168.40.1.5351: UDP, length 24
19:37:55.300027 IP 192.168.40.39.55225 > 192.168.40.1.1900: UDP, length 94
19:37:55.300044 IP 192.168.40.39.55225 > 239.255.255.250.1900: UDP, length 94
19:37:55.554108 IP 192.168.40.39.43445 > 203.0.113.1.12345: UDP, length 73
19:37:55.554203 IP 192.168.40.39.58064 > 192.168.40.1.5351: UDP, length 2
19:37:55.554231 IP 192.168.40.39.58064 > 192.168.40.1.5351: UDP, length 24
19:37:55.554252 IP 192.168.40.39.58064 > 192.168.40.1.1900: UDP, length 94
19:37:55.554271 IP 192.168.40.39.58064 > 239.255.255.250.1900: UDP, length 94
19:37:55.807041 IP 192.168.40.39.36578 > 203.0.113.1.12345: UDP, length 73
19:37:55.807394 IP 192.168.40.39.54733 > 192.168.40.1.5351: UDP, length 2
19:37:55.807416 IP 192.168.40.39.54733 > 192.168.40.1.5351: UDP, length 24
19:37:55.807441 IP 192.168.40.39.54733 > 192.168.40.1.1900: UDP, length 94
19:37:55.807465 IP 192.168.40.39.54733 > 239.255.255.250.1900: UDP, length 94
19:37:56.301656 IP 192.168.40.39.35667 > 203.0.113.1.12345: UDP, length 73
19:37:56.301777 IP 192.168.40.39.51857 > 192.168.40.1.5351: UDP, length 2
19:37:56.301817 IP 192.168.40.39.51857 > 192.168.40.1.5351: UDP, length 24
19:37:56.301847 IP 192.168.40.39.51857 > 192.168.40.1.1900: UDP, length 94
19:37:56.301876 IP 192.168.40.39.51857 > 239.255.255.250.1900: UDP, length 94
19:37:56.555142 IP 192.168.40.39.49900 > 203.0.113.1.12345: UDP, length 73
19:37:56.555278 IP 192.168.40.39.36065 > 192.168.40.1.5351: UDP, length 2
19:37:56.555315 IP 192.168.40.39.36065 > 192.168.40.1.5351: UDP, length 24
19:37:56.555335 IP 192.168.40.39.36065 > 192.168.40.1.1900: UDP, length 94
19:37:56.555351 IP 192.168.40.39.36065 > 239.255.255.250.1900: UDP, length 94
19:37:56.806897 IP 192.168.40.39.55695 > 203.0.113.1.12345: UDP, length 73
19:37:56.807027 IP 192.168.40.39.55198 > 192.168.40.1.5351: UDP, length 2
19:37:56.807064 IP 192.168.40.39.55198 > 192.168.40.1.5351: UDP, length 24
19:37:56.807099 IP 192.168.40.39.55198 > 192.168.40.1.1900: UDP, length 94
19:37:56.807118 IP 192.168.40.39.55198 > 239.255.255.250.1900: UDP, length 94
19:37:57.303374 IP 192.168.40.39.34384 > 203.0.113.1.12345: UDP, length 73
19:37:57.303460 IP 192.168.40.39.43947 > 192.168.40.1.5351: UDP, length 2
19:37:57.303487 IP 192.168.40.39.43947 > 192.168.40.1.5351: UDP, length 24
19:37:57.303511 IP 192.168.40.39.43947 > 192.168.40.1.1900: UDP, length 94
19:37:57.303533 IP 192.168.40.39.43947 > 239.255.255.250.1900: UDP, length 94
^C
45 packets captured
45 packets received by filter
0 packets dropped by kernel
```

Log excerpt from `tailscaled -verbose 1`:
```
LinkChange: minor
[RATELIMIT] format("LinkChange: minor")
magicsock: starting endpoint update (link-change-minor)
[RATELIMIT] format("magicsock: starting endpoint update (%s)")
[RATELIMIT] format("netcheck: report: %v") (28 dropped)
netcheck: report: udp=true v6=true mapvarydest=false hair=false portmap= v4a=<v4 address>:55878 v6a=[<v6 address>]:41178 derp=3 derpdist=3v4:6ms,3v6:7ms,6v4:42ms,7v4:70ms
[RATELIMIT] format("magicsock: ReSTUN: endpoint update active, need another later (%q)") (14 dropped)
magicsock: ReSTUN: endpoint update active, need another later ("link-change-minor")
netcheck: report: udp=true v6=true mapvarydest=false hair=false portmap= v4a=<v4 address>:55878 v6a=[<v6 address>]:41178 derp=3 derpdist=3v4:7ms,3v6:6ms,6v4:42ms,7v4:70ms
[RATELIMIT] format("netcheck: report: %v")
magicsock: ReSTUN: endpoint update active, need another later ("link-change-minor")
[RATELIMIT] format("magicsock: ReSTUN: endpoint update active, need another later (%q)")
[RATELIMIT] format("LinkChange: minor") (28 dropped)
```

I don't think it breaks anything, but this seems like an issue. IMHO there should be a backoff mechanism.

### Steps to reproduce

No idea. I just noticed this.

### OS

Linux

### OS version

Arch Linux

### Tailscale version

1.22.2

### Bug report

BUG-e0c14d268118970547b2e76725e2a708a3e3faa5d531e7c6fb54dd51a7b7f6fd-20220320113733Z-1e3ccb2a744c4b07

**Comment by hexchain** [comment]
Looks like the root cause is the netlink monitor reporting that IP addresses have changed:

```
% ip -o monitor address | ts
Mar 20 20:23:39 2: eno1    inet 192.168.40.39/24 brd 192.168.40.255 scope global dynamic noprefixroute eno1\       valid_lft 34743sec preferred_lft 34743sec
Mar 20 20:23:39 2: eno1    inet6 <v6 address>/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Mar 20 20:23:39 2: eno1    inet6 <v6 address>/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 67882sec preferred_lft 67882sec
Mar 20 20:23:39 2: eno1    inet6 <v6 address>/64 scope global temporary dynamic \       valid_lft 67882sec preferred_lft 67882sec
Mar 20 20:23:39 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67882sec preferred_lft 0sec
Mar 20 20:23:39 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67882sec preferred_lft 0sec
Mar 20 20:23:39 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67882sec preferred_lft 0sec
Mar 20 20:23:40 2: eno1    inet 192.168.40.39/24 brd 192.168.40.255 scope global dynamic noprefixroute eno1\       valid_lft 34743sec preferred_lft 34743sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 67882sec preferred_lft 67882sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary dynamic \       valid_lft 67882sec preferred_lft 67882sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67882sec preferred_lft 0sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67882sec preferred_lft 0sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67882sec preferred_lft 0sec
Mar 20 20:23:40 2: eno1    inet 192.168.40.39/24 brd 192.168.40.255 scope global dynamic noprefixroute eno1\       valid_lft 34742sec preferred_lft 34742sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 67881sec preferred_lft 67881sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary dynamic \       valid_lft 67881sec preferred_lft 67881sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67881sec preferred_lft 0sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67881sec preferred_lft 0sec
Mar 20 20:23:40 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67881sec preferred_lft 0sec
Mar 20 20:23:41 2: eno1    inet 192.168.40.39/24 brd 192.168.40.255 scope global dynamic noprefixroute eno1\       valid_lft 34742sec preferred_lft 34742sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 67881sec preferred_lft 67881sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary dynamic \       valid_lft 67881sec preferred_lft 67881sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamc \       valid_lft 67881sec preferred_lft 0sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67881sec preferred_lft 0sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67881sec preferred_lft 0sec
Mar 20 20:23:41 2: eno1    inet 192.168.40.39/24 brd 192.168.40.255 scope global dynamic noprefixroute eno1\       valid_lft 34741sec preferred_lft 34741sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 67880sec preferred_lft 67880sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary dynamic \       valid_lft 67880sec preferred_lft 67880sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67880sec preferred_lft 0sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67880sec preferred_lft 0sec
Mar 20 20:23:41 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67880sec preferred_lft 0sec
Mar 20 20:23:42 2: eno1    inet 192.168.40.39/24 brd 192.168.40.255 scope global dynamic noprefixroute eno1\       valid_lft 34741sec preferred_lft 34741sec
Mar 20 20:23:42 2: eno1    inet6 <v6 address>/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Mar 20 20:23:42 2: eno1    inet6 <v6 address>/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 67880sec preferred_lft 67880sec
Mar 20 20:23:42 2: eno1    inet6 <v6 address>/64 scope global temporary dynamic \       valid_lft 67880sec preferred_lft 67880sec
Mar 20 20:23:42 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67880sec preferred_lft 0sec
Mar 20 20:23:42 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67880sec preferred_lft 0sec
Mar 20 20:23:42 2: eno1    inet6 <v6 address>/64 scope global temporary deprecated dynamic \       valid_lft 67880sec preferred_lft 0sec
```

However, it seems `valid_lft`s are the only difference.

**Comment by bradfitz** [comment]
Good find on valid_lft! I'll investigate later.

**Comment by hexchain** [comment]
How is your IPv6 address assigned? In my case, the fiber provider assigns a /64 prefix through DHCP-PD, and my router announces this prefix using SLAAC (dnsmasq). My guess is that the valid lifetime has something to do with the DHCP lease time.

**Comment by raggi** [comment]
I have a very similar setup with v6-pd into my network from dnsmasq. I've not seen preferred lifetime adverts with 0s, however I did have a bug in my configuration the first time I was setting things up where I ended up with two advertising routers on my network. That lead to a situation that had what were probably similar or related symptoms, though I don't know offhand if they produced exactly this scenario.

Is it possible that you have two routers advertising addresses?

**Comment by hexchain** [comment]
I don't think so. I only have one router (hardware), and on it only dnsmasq should be sending RAs.

`rdisc6 -m eno1` also shows only one response:
```
% sudo rdisc6 -m eno1
Soliciting ff02::2 (ff02::2) on eno1...

Hop limit                 :           64 (      0x40)
Stateful address conf.    :           No
Stateful other conf.      :           No
Mobile home agent         :           No
Router preference         :       medium
Neighbor discovery proxy  :           No
Router lifetime           :         1800 (0x00000708) seconds
Reachable time            :  unspecified (0x00000000)
Retransmit time           :  unspecified (0x00000000)
 Prefix                   : <v6 prefix>::/64
  On-link                 :          Yes
  Autonomous address conf.:          Yes
  Valid time              :        48539 (0x0000bd9b) seconds
  Pref. time              :        48539 (0x0000bd9b) seconds
 MTU                      :         1500 bytes (valid)
 Source link-layer address: <router macaddr>
 DNS search list          : <dns search domain>
  DNS search list lifetime:        48539 (0x0000bd9b) seconds
 Recursive DNS server     : <router v6 address>
  DNS server lifetime     :        48539 (0x0000bd9b) seconds
 from fe80::<router lladdr>
```

The "Valid/Pref. time" values of 48539 roughly match the `valid_lft` values of the current global ipv6 addresses.

**Comment by raggi** [comment]
Ok, good, thank you. That was important to check.

The next thing to check is if the system is configured to use IPv6 privacy extensions, here:

```sh
sysctl net.ipv6 | grep temp
```

This may produce something like the following output:

```
net.ipv6.conf.all.temp_prefered_lft = 86400
net.ipv6.conf.all.temp_valid_lft = 604800
net.ipv6.conf.all.use_tempaddr = 1
net.ipv6.conf.default.temp_prefered_lft = 86400
net.ipv6.conf.default.temp_valid_lft = 604800
net.ipv6.conf.default.use_tempaddr = 1
```

The `use_tempaddr` being `1` would lead to this situation. If this is the case, I'd love to know if this is a default in your distribution, and which distribution if you don't mind saying.

**Comment by hexchain** [comment]
Yes, I do have privacy extensions turned on:

```
% sudo sysctl net.ipv6 | grep temp
net.ipv6.conf.all.temp_prefered_lft = 86400
net.ipv6.conf.all.temp_valid_lft = 604800
net.ipv6.conf.all.use_tempaddr = 0
net.ipv6.conf.default.temp_prefered_lft = 86400
net.ipv6.conf.default.temp_valid_lft = 604800
net.ipv6.conf.default.use_tempaddr = 0
net.ipv6.conf.eno1.temp_prefered_lft = 86400
net.ipv6.conf.eno1.temp_valid_lft = 604800
net.ipv6.conf.eno1.use_tempaddr = 2
net.ipv6.conf.lo.temp_prefered_lft = 86400
net.ipv6.conf.lo.temp_valid_lft = 604800
net.ipv6.conf.lo.use_tempaddr = -1
net.ipv6.conf.tailscale0.temp_prefered_lft = 86400
net.ipv6.conf.tailscale0.temp_valid_lft = 604800
net.ipv6.conf.tailscale0.use_tempaddr = -1
net.ipv6.conf.virbr0.temp_prefered_lft = 86400
net.ipv6.conf.virbr0.temp_valid_lft = 604800
net.ipv6.conf.virbr0.use_tempaddr = 0
net.ipv6.conf.virbr1.temp_prefered_lft = 86400
net.ipv6.conf.virbr1.temp_valid_lft = 604800
net.ipv6.conf.virbr1.use_tempaddr = 0
net.ipv6.conf.wlan0.temp_prefered_lft = 86400
net.ipv6.conf.wlan0.temp_valid_lft = 604800
net.ipv6.conf.wlan0.use_tempaddr = 0
```

I'm using Arch Linux, but I believe it is not a distro setting. NetworkManager has the option to enable and prefer temporary addresses:
```
% nmcli connection show Wired\ Auto | grep priv
ipv6.ip6-privacy:                       2 (enabled, prefer temporary IP)
ipv6.addr-gen-mode:                     stable-privacy
```

**Comment by DentonGentry** [comment]
I believe this is fixed by https://github.com/tailscale/tailscale/issues/4282

**Comment by hexchain** [comment]
I don't think it's already fixed though? But I also don't mind closing this in favor of #4282, which contains more investigation and is better described.

**Comment by bradfitz** [comment]
Reopening for now.

**Comment by raggi** [comment]
@hexchain I believe this is fixed in the unstable builds available at https://pkgs.tailscale.com, though as there are multiple potential causes I would love to get verification before closing. If you're able to try an unstable release, that would be great, otherwise you could also wait for 1.24 that should include this fix.

**Comment by heyheyhello** [comment]
Installed an unstable build. It...helped. It used to be an SSDP flood every 3 seconds and now on the unstable build it's every 30 seconds.

Also the old floods were a lot more packets. Ran Wireshark for 5 minutes before and after updating to unstable:

- Before: ~4400 SSDP packets in 5m (48% of all laptop traffic on idle).
- After: ~360 SSDP packets in 5m (6% of all traffic on idle).

Obviously not the cleanest test; it's possible Firefox tabs or something was running. But even just looking at the timestamps of the packets there seems to be a lot less activity in the unstable build:

**Before:**
![image](https://user-images.githubusercontent.com/44614862/165045196-231f17e5-6a0e-4090-b9e2-aa7b900aa5db.png)

**After (Unstable build):**
![image](https://user-images.githubusercontent.com/44614862/165045582-892d356c-bba4-49fa-bd3c-a388b08047e0.png)

Not to get too off topic but I wish I could leave Tailscale on and not think about it; I can't. Even with the SSDP issues aside, I see it eat a pretty constant stream of data (usually between 0.5kb/s to 6kb/s) forever... :/

**Comment by raggi** [comment]
Thank you @heyheyhello. I will look again into the SSDP path and see if there are other possible causes of it doing extra work. Can you tell me if `ip monitor address` is regularly showing changes on your system? The patch I applied for 1.24 will ignore regular changes that _don't change addresses_, but if your addresses are changing regularly that would be a different story - I'd like to rule this out as a cause.

Regarding seeing status flip, part of that is normal, and part of it may not be. The normal part is that direct connections may take time to establish - they're established on-demand. What is not normal is to see that flapping back to relay regularly.

**Comment by heyheyhello** [comment]
I see a lot of output but the address isn't changing:

<details>

<summary>Output</summary>

```
❯ ip -o monitor address | ts
Apr 25 10:57:52 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:57:52 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:57:52 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86169sec
Apr 25 10:57:56 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:57:56 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:57:56 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86166sec
Apr 25 10:58:00 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:00 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:00 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86162sec
Apr 25 10:58:03 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:03 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:03 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86159sec
Apr 25 10:58:07 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:07 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:07 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86155sec
Apr 25 10:58:11 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:11 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:11 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86151sec
Apr 25 10:58:14 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:14 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:14 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86148sec
Apr 25 10:58:18 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:18 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:18 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86144sec
Apr 25 10:58:22 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:22 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:22 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86140sec
Apr 25 10:58:26 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:26 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:26 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86136sec
Apr 25 10:58:29 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:29 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:29 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86133sec
Apr 25 10:58:33 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:33 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:33 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86129sec
Apr 25 10:58:37 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:37 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:37 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86125sec
Apr 25 10:58:40 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:40 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:40 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86121sec
Apr 25 10:58:44 5: wlp3s0    inet6 fe80::2225:6f78:f2a0:848/64 scope link noprefixroute \       valid_lft forever preferred_lft forever
Apr 25 10:58:44 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002/64 scope global dynamic mngtmpaddr noprefixroute \       valid_lft 535429sec preferred_lft 401572sec
Apr 25 10:58:44 5: wlp3s0    inet6 fd00:a84e:3f4f:5ee2:204f:ac9a:eeac:7719/64 scope global temporary dynamic \       valid_lft 535429sec preferred_lft 86118sec
```

</details>

**Comment by raggi** [comment]
I just saw the reproduction steps too, that's extremely useful thank you again! A good deal of this sounds normal, but also likely that there's scope for improvement. The volume of packets shown in your Wireshark capture looks like a lot, but I'll need to check on the protocol requirements.

**Comment by heyheyhello** [comment]
Yes it's working:

```
~ 
❯ systemctl restart tailscaled

~ 
❯ sudo tailscale status
Tailscale is stopped.

~ 
❯ journalctl -u tailscaled -f
-- Journal begins at Thu 2021-10-14 22:33:49 PDT. --
Apr 25 11:19:33 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002 / <nil>
Apr 25 11:19:33 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s")
Apr 25 11:19:36 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s") (4 dropped)
Apr 25 11:19:36 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fe80::2225:6f78:f2a0:848
Apr 25 11:19:36 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002
Apr 25 11:19:36 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s")
Apr 25 11:19:44 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s") (7 dropped)
Apr 25 11:19:44 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fe80::2225:6f78:f2a0:848 / <nil>
Apr 25 11:19:44 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002 / <nil>
Apr 25 11:19:44 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s")
Apr 25 11:19:47 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s") (7 dropped)
Apr 25 11:19:47 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fe80::2225:6f78:f2a0:848
Apr 25 11:19:47 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002
Apr 25 11:19:47 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s")
Apr 25 11:19:54 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s") (7 dropped)
Apr 25 11:19:54 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fe80::2225:6f78:f2a0:848 / <nil>
Apr 25 11:19:54 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002 / <nil>
Apr 25 11:19:54 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s")
Apr 25 11:19:57 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s") (7 dropped)
Apr 25 11:19:57 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fe80::2225:6f78:f2a0:848
Apr 25 11:19:57 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002
Apr 25 11:19:57 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s")
Apr 25 11:20:05 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s") (7 dropped)
Apr 25 11:20:05 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fe80::2225:6f78:f2a0:848 / <nil>
Apr 25 11:20:05 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002 / <nil>
Apr 25 11:20:05 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s")
Apr 25 11:20:09 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s") (7 dropped)
Apr 25 11:20:09 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fe80::2225:6f78:f2a0:848
Apr 25 11:20:09 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002
Apr 25 11:20:09 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s")
Apr 25 11:20:12 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s") (4 dropped)
Apr 25 11:20:12 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fe80::2225:6f78:f2a0:848 / <nil>
Apr 25 11:20:12 twothirty tailscaled[46493]: monitor: RTM_NEWADDR: wlp3s0(5) fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002 / <nil>
Apr 25 11:20:12 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: %s: %s(%d) %s / %s")
Apr 25 11:20:16 twothirty tailscaled[46493]: [RATELIMIT] format("monitor: ignored duplicate RTM_NEWADDR for %s") (4 dropped)
Apr 25 11:20:16 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fe80::2225:6f78:f2a0:848
Apr 25 11:20:16 twothirty tailscaled[46493]: monitor: ignored duplicate RTM_NEWADDR for fd00:a84e:3f4f:5ee2:d845:cade:1bef:f002
...
...
```

Runs forever with Tailscale stopped (daemon is running but `tailscale status` says stopped). I then started it, saw a bit of SSDP, then nothing, then I ran `ping` at around 11:22:10 to kick off the direct link. Saw "Accept: TCP{...}" a few times in the logs. Saw SSDP every 30s or so in Wireshark. I guess there are no verbose logs for SSDP?

**Comment by timaelliott** [comment]
This is still happening on 1.44.0 when I'm on a stable direct connection to my node. Please fix this.

**Comment by tkbstudios** [comment]
same here

**Comment by DevDorrejo** [comment]
I am having the same issue, false positive in firewall fortigate

**Comment by donadelden** [comment]
Same problem identified by Suricata as `ET DOS Possible SSDP Amplification Scan in Progress`

**Comment by Hl2run** [comment]
Same here with fortigate alerts going off.

**Comment by rasmusson** [comment]
Seeing the same thing deployed on kubernetes

**Comment by jsa91** [comment]
+1 
Suricata reports `SSDP Amplification Scan in Progress`

**Comment by mileyceberus** [comment]
Hello, just advising that this issue still persists on opnsense-26.1.2-amd64 running tailscale-plugin-1.3. To recap, 

1. SSDP flood seen on WAN interface (dst 239.255.255.255:1900) when tailscale is running.
2. SSDP flood stops when tailscale stops.

Thanks & regards..
